Use the MediaUploadCheck component before each Upload component

imath opened the pull request:
This PR should fix #11910

## Description
1. It adds a MediaUploadCheck component before the forgotten MediaUpload components used directly in some blocks :
- within the edit toolbar buttons of the cover, file and image blocks,
- within the Poster image setting of the video block.
2. It also adds a MediaUploadCheck component before rendering the Block Drop Zone.

## How has this been tested?
Using WordPress trunk & Gutenberg Master I've visually checked the bugs listed into the #11910 issue were fixed for the contributor role (user without the `upload_files` capability). I also ran successfully the unit tests suite.

## Types of changes
Improves #4155 by making sure the forgotten Upload components are used after the MediaUploadCheck one.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

imath committed:
Use the MediaUploadCheck component before each Upload component

This PR should fix #11910

1. It adds a MediaUploadCheck component before the forgotten MediaUpload components used directly in some blocks :
- within the edit toolbar buttons of the cover, file and image blocks,
- within the Poster image setting of the video block.
2. It also adds a MediaUploadCheck component before rendering the Block Drop Zone.

gziolo commented:
It tests well. Great work on fixing all the remaining issues.

We discussed it on Slack, but I wanted to double check if you still think it is an option to include `MediaUploadCheck` inside `MediaUpload` component to ensure that it is always disabled when a user doesn't have proper capabilities? I really like how it is done for `DropZone` and `MediaPlaceholder`.

imath committed:
Edit the MediaUpload documentation about Upload Permissions

Include some informations about how to make sure the current user has upload permissions (by wrapping the MediaUpload component into the MediaUploadCheck one)

imath commented:
@gziolo I've been looking into it and came to the conclusion that it's fine the way it is actually 😃 . Into the MediaPlaceHolder we have the MediaUpload component but we also have the DropZone and the FormFileUpload components. So I've tried to do the same UploadPermissions check into these components but never succeeded : it looks like the upload permissions are not set yet.

So I thought, as we need to wrap the DropZone, FormFileUpload and the MediaUpload components into a MediaUploadCheck one anyway, doing an extra check into MediaUpload was weird.

gziolo commented:
Let's go with the current approach and see how it work. There are so many different use cases where the check for upload permissions is necessary that it's hard to find a perfect approach. We can always iterate on it. Thanks for fixing all remaining bugs I could find.
